test/correctness/bitcast_vector_load_into_scalar.cpp

```diff
@@ -0,0 +1,140 @@
+    buf.for_each_value([&](CHUNK_TYPE &v) { v = dist(gen); });
+
+    return buf;
+}
+
+namespace {
+enum class ChunkOrder {
+    LowestFirst,
+    HighestFirst,
+};
+};
+
+template<typename CHUNK_TYPE>
+bool test_with_n_times_chunk_type(int num_chunks, Target t) {
+    const int width = 256 * num_chunks;
+    Buffer<CHUNK_TYPE> input_buf = gen_random_chunks<CHUNK_TYPE>({width});
+
+    using WIDE_STORAGE_TYPE = uint64_t;
+    const int CHUNK_WIDTH = 8 * sizeof(CHUNK_TYPE);
+    const int WIDE_TYPE_WIDTH = CHUNK_WIDTH * num_chunks;
+
+    Var x("x");
+
+    int wide_width = width / num_chunks;
+
+    auto forward = [wide_width, num_chunks, x, WIDE_TYPE_WIDTH,
+                    t](const Func &input, ChunkOrder chunk_order) {
+        Buffer<WIDE_STORAGE_TYPE> wide(wide_width);
+        std::vector<Expr> input_chunks;
+        for (int i = 0; i != num_chunks; ++i)
+            input_chunks.push_back(input((num_chunks * x) + i));
+        if (chunk_order != ChunkOrder::LowestFirst)
+            std::reverse(input_chunks.begin(), input_chunks.end());
+        Func widen("widened");
+        widen(x) = reinterpret(UInt(WIDE_TYPE_WIDTH),
+                               Internal::Shuffle::make_concat(input_chunks));
+        Func store("store");
+        store(x) = cast<WIDE_STORAGE_TYPE>(widen(x));
+        store.realize(wide, t);
+        return wide;
+    };
+
+    auto forward_naive = [wide_width, num_chunks](Buffer<CHUNK_TYPE> input_buf,
+                                                  ChunkOrder chunk_order) {
+        Buffer<WIDE_STORAGE_TYPE> wide(wide_width);
+        for (int32_t x = 0; x < wide_width; x++) {
+            WIDE_STORAGE_TYPE &v = wide(x);
+            v = 0;
+            for (int chunk = 0; chunk != num_chunks; ++chunk) {
+                int chunk_idx = chunk_order == ChunkOrder::HighestFirst ?
+                                    chunk :
+                                    (num_chunks - 1) - chunk;
+                v <<= CHUNK_WIDTH;
+                v |= (WIDE_STORAGE_TYPE)input_buf(num_chunks * x + chunk_idx);
+            }
+        }
+        return wide;
+    };
+
+    Func input("input");
+    input(x) = input_buf(x);
+
+    bool success = true;
+    for (ChunkOrder chunk_order :
+         {ChunkOrder::LowestFirst, ChunkOrder::HighestFirst}) {
+        const auto wide_actual = forward(input, chunk_order);
+        const auto wide_expected = forward_naive(input_buf, chunk_order);
+        success &= expect_eq(wide_actual, wide_expected);
+    }
+
+    return success;
+}
+
+template<typename CHUNK_TYPE>
+bool test_with_chunk_type(Target t) {
+    bool success = true;
+
+    const int CHUNK_WIDTH = 8 * sizeof(CHUNK_TYPE);
+    for (int num_chunks = 2; CHUNK_WIDTH * num_chunks <= 64; num_chunks *= 2) {
+        success &= test_with_n_times_chunk_type<CHUNK_TYPE>(num_chunks, t);
+    }
+
+    return success;
+}
+
+bool test_all(Target t) {
+    bool success = true;
+
+    success &= test_with_chunk_type<uint8_t>(t);
```

Comment:
There is some kind of failure to simplify for the i8 chunk type somewhere in halide:
```
   let t5 = b0[(store.min.0 + store.s0.x.rebased)*2]
   store[store.s0.x.rebased] = uint64((uint16)reinterpret(ramp(t5, b0[((store.min.0 + store.s0.x.rebased)*2) + 1] - t5, 2)))
```
but suddenly it's fine for larger types:
```
   store$1[store$1.s0.x.rebased] = uint64((uint32)reinterpret(b3[ramp((store$1.min.0 + store$1.s0.x.rebased)*4, 1, 4) aligned(4, 0)]))
```